cobalt/web/agent.cc

```diff
@@ -539,6 +539,8 @@ void Agent::Stop() {
 
   // Ensure that the destruction observer got added before stopping the thread.
   destruction_observer_added_.Wait();
+  // Wait for all previously posted tasks to finish.
+  thread_.message_loop()->task_runner()->WaitForFence();
   // Stop the thread. This will cause the destruction observer to be notified.
   thread_.Stop();
```

Comment:
yes I think it can be reset here too, I'll do some checks first, it may be that the object can be referenced from somewhere else even if the thread isn't running anymore, so there may be a need to change other code to change this to reset.

Reply:
Oh, this is not a unique_ptr, it's a regular member, so can't be reset.